src/main/datatypes/MQ2GroupType.cpp

```diff
@@ -15,768 +15,1125 @@
 
-	int nMembers = pLocalPC->Group->GetNumberOfMembersExcludingSelf();
+	enum class GroupMembers
+	{
+		Member = 1,
+		Members,
+		Leader,
+		GroupSize,
+		MainTank,
+		MainAssist,
+		Puller,
+		MarkNpc,
+		MasterLooter,
+		AnyoneMissing,
+		Present,
+		MercenaryCount,
+		TankMercCount,
+		HealerMercCount,
+		MeleeMercCount,
+		CasterMercCount,
+		MouseOver,
+		AvgHPs,
+		Injured,
+		LowMana,
+		Cleric,
+		Shaman,
+		Warrior,
+		Paladin,
+		Ranger,
+		Shadowknight,
+		Druid,
+		Monk,
+		Bard,
+		Rogue,
+		Necromancer,
+		Wizard,
+		Mage,
+		Magician,
+		Enchanter,
+		Beastlord,
+		Berserker,
+		Furthest,
+		FurthestMelee,
+		FurthestPureCaster
+	};
+
+	MQ2GroupType::MQ2GroupType() : MQ2Type( "group" )
+	{
+		ScopedTypeMember( GroupMembers, Member );
+		ScopedTypeMember( GroupMembers, Members );
+		ScopedTypeMember( GroupMembers, Leader );
+		ScopedTypeMember( GroupMembers, GroupSize );
+		ScopedTypeMember( GroupMembers, MainTank );
+		ScopedTypeMember( GroupMembers, MainAssist );
+		ScopedTypeMember( GroupMembers, Puller );
+		ScopedTypeMember( GroupMembers, MarkNpc );
+		ScopedTypeMember( GroupMembers, MasterLooter );
+		ScopedTypeMember( GroupMembers, AnyoneMissing );
+		ScopedTypeMember( GroupMembers, Present );
+		ScopedTypeMember( GroupMembers, MercenaryCount );
+		ScopedTypeMember( GroupMembers, TankMercCount );
+		ScopedTypeMember( GroupMembers, HealerMercCount );
+		ScopedTypeMember( GroupMembers, MeleeMercCount );
+		ScopedTypeMember( GroupMembers, CasterMercCount );
+		ScopedTypeMember( GroupMembers, MouseOver );
+		ScopedTypeMember( GroupMembers, AvgHPs );
+		ScopedTypeMember( GroupMembers, Injured );
+		ScopedTypeMember( GroupMembers, LowMana );
+		ScopedTypeMember( GroupMembers, Cleric );
+		ScopedTypeMember( GroupMembers, Shaman );
+		ScopedTypeMember( GroupMembers, Warrior );
+		ScopedTypeMember( GroupMembers, Paladin );
+		ScopedTypeMember( GroupMembers, Ranger );
+		ScopedTypeMember( GroupMembers, Shadowknight );
+		ScopedTypeMember( GroupMembers, Druid );
+		ScopedTypeMember( GroupMembers, Monk );
+		ScopedTypeMember( GroupMembers, Bard );
+		ScopedTypeMember( GroupMembers, Rogue );
+		ScopedTypeMember( GroupMembers, Necromancer );
+		ScopedTypeMember( GroupMembers, Wizard );
+		ScopedTypeMember( GroupMembers, Mage );
+		ScopedTypeMember( GroupMembers, Magician );
+		ScopedTypeMember( GroupMembers, Enchanter );
+		ScopedTypeMember( GroupMembers, Beastlord );
+		ScopedTypeMember( GroupMembers, Berserker );
+		ScopedTypeMember( GroupMembers, Furthest );
+		ScopedTypeMember( GroupMembers, FurthestMelee );
+		ScopedTypeMember( GroupMembers, FurthestPureCaster );
 
-	_itoa_s(nMembers, Destination, MAX_STRING, 10);
-	return true;
-}
+	}
 
-bool MQ2GroupType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest)
-{
-	MQTypeMember* pMember = MQ2GroupType::FindMember(Member);
+	bool MQ2GroupType::ToString( MQVarPtr VarPtr, char* Destination )
+	{
+		if( !pLocalPC || !pLocalPC->Group ) return false;
```

Comment:
what is with all the extra spaces around brakcets

Reply:
It looks like this got run through a formatter, which is making it hard to review (and makes the style inconsistent with the rest of the project). Please limit changes to actual code changes

Reply:
ya sorry my VS settings copy / paste auto updates formatting. I can reform it.